pkg/vsphere/tasks/waiter.go

```diff
@@ -89,11 +89,26 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 	}
 }
 
-func isTaskInProgress(err error) bool {
+const (
+	vimFault  = "vim"
+	soapFault = "soap"
+	taskFault = "task"
+)
+
+// isRetryErrors will return true for vSphere errors, which can be fixed by retry.
+// Currently the error includes TaskInProgress, NetworkDisruptedAndConfigRolledBack and InvalidArgument
+// The reason to retry for these errors is because sometimes vSphere needs longer time to finish sync up inside
+func isRetryErrors(err error) bool {
 	if soap.IsSoapFault(err) {
 		switch f := soap.ToSoapFault(err).VimFault().(type) {
 		case types.TaskInProgress:
 			return true
+		case *types.NetworkDisruptedAndConfigRolledBack:
+			logExpectedFault(soapFault, f)
+			return true
+		case *types.InvalidArgument:
```

Comment:
I would guess this is because before the vsan converges, we are passing an invalid path to it.  Unfortunate return code in my opinion, but what we have to deal with...